packages/sw-runtime-caching/src/lib/request-wrapper.js

```diff
@@ -74,34 +74,31 @@ class RequestWrapper {
       this.matchOptions = matchOptions;
     }
 
-    this.pluginCallbacks = {};
+    this.plugins = new Map();
 
     if (plugins) {
       assert.isArrayOfType({plugins}, 'object');
 
       plugins.forEach((plugin) => {
         for (let callbackName of pluginCallbacks) {
           if (typeof plugin[callbackName] === 'function') {
-            if (!this.pluginCallbacks[callbackName]) {
-              this.pluginCallbacks[callbackName] = [];
+            if (!this.plugins.has(callbackName)) {
+              this.plugins.set(callbackName, []);
             }
-            this.pluginCallbacks[callbackName].push(
-              plugin[callbackName].bind(plugin));
+            this.plugins.get(callbackName).push(plugin);
           }
         }
       });
     }
 
-    if (this.pluginCallbacks.cacheWillUpdate) {
-      if (this.pluginCallbacks.cacheWillUpdate.length !== 1) {
-        throw ErrorFactory.createError('multiple-cache-will-update-plugins');
-      }
+    if (this.plugins.has('cacheWillUpdate') &&
+        this.plugins.get('cacheWillUpdate').length !== 1) {
```

Comment:
Should we block on being able to add multiple cacheWillUpdate plugins rather than add them and then throw afterwards?